src/main/java/org/apache/maven/plugins/shade/DefaultShader.java

```diff
@@ -256,6 +256,10 @@ else if ( shadeRequest.isShadeSourcesContent() && name.endsWith( ".java" ) )
 
                     addResource( resources, jos, mappedName, entry.getTime(), in );
                 }
+                else
+                {
+                    duplicates.remove( name, jar );
```

Comment:
Could there be a test for this?

Reply:
Added https://github.com/apache/maven-shade-plugin/pull/48/files#diff-bc8729c3b7ec6843be39d656cb9b2d40R96